services/workflows-service/src/auth/workflow-token/workflow-token.repository.ts

```diff
@@ -24,16 +24,25 @@ export class WorkflowTokenRepository {
   }
 
   async findByTokenUnscoped(token: string) {
-    return await this.prisma.workflowRuntimeDataToken.findFirst({
+    return await this.prismaService.workflowRuntimeDataToken.findFirst({
       where: {
         token,
         AND: [{ expiresAt: { gt: new Date() } }, { deletedAt: null }],
       },
     });
   }
 
+  async findByTokenWithExpiredUnscoped(token: string) {
+    return await this.prismaService.workflowRuntimeDataToken.findFirst({
+      where: {
+        token,
```

Comment:
Since this table will queried based on 'token' field in order to get related context, we might want to add an index on top of this field